pkg/telemetryquerylanguage/tql/parser.go

```diff
@@ -25,15 +26,54 @@ import (
 // ParsedQuery represents a parsed query. It is the entry point into the query DSL.
 // nolint:govet
 type ParsedQuery struct {
-	Invocation Invocation `@@`
-	Condition  *Condition `( "where" @@ )?`
+	Invocation  Invocation         `@@`
+	WhereClause *BooleanExpression `( "where" @@ )?`
 }
 
-// Condition represents an optional boolean condition on the RHS of a query.
+// BooleanValue represents something that evaluates to a boolean --
+// either an equality or inequality, explicit true or false, or
+// a parenthesized subexpression.
+// nolint:govet
+type BooleanValue struct {
+	Condition *Condition         `( @@`
+	ConstExpr *Boolean           `| @Boolean`
+	SubExpr   *BooleanExpression `| "(" @@ ")" )`
+}
+
+// OpBooleanValue represents the right side of an AND boolean expression.
+// nolint:govet
+type OpBooleanValue struct {
```

Comment:
Can we rename this to indicate it is always part of an `AND` clause?  `OpTerm` below is always part of an `OR` clause, so the `Op` part of the name doesn't help identify what the struct represents in the grammar.  Maybe `OpAndBooleanValue` and `OpOrTerm`

Reply:
I had considered Term and Factor, but I couldn't keep them straight in my head, so yeah, let's try your suggestions.